File: crates/marmot-uniffi/tests/details_markdown_smoke.kt
package dev.ipf.marmotkit

fun main() {
    val block = MarkdownBlockFfi.Details(
        summary = listOf(MarkdownInlineFfi.Text("More")),
        open = false,
        body = listOf(MarkdownBlockFfi.Paragraph(listOf(MarkdownInlineFfi.Text("body")))),
        blankLinesBefore = byteArrayOf(1),
    )
    val copy = FfiConverterTypeMarkdownBlockFfi.lift(FfiConverterTypeMarkdownBlockFfi.lower(block))
    check(copy is MarkdownBlockFfi.Details)
    check(copy.summary == block.summary && copy.open == block.open && copy.body == block.body)
    check(copy.blankLinesBefore.contentEquals(block.blankLinesBefore))
    println("Kotlin Markdown Details DTO round trip passed")
}

File: crates/marmot-uniffi/tests/details_markdown_smoke.swift
import Foundation

@main
struct DetailsMarkdownSmoke {
    static func main() throws {
        let block = MarkdownBlockFfi.details(
            summary: [MarkdownInlineFfi.text(content: "More")],
            open: false,
            body: [MarkdownBlockFfi.paragraph(inlines: [MarkdownInlineFfi.text(content: "body")])],
            blankLinesBefore: Data([1])
        )
        let copy = try FfiConverterTypeMarkdownBlockFfi.lift(FfiConverterTypeMarkdownBlockFfi.lower(block))
        precondition(copy == block)
        print("Swift Markdown Details DTO round trip passed")
    }
}
